aws/resource_aws_wafregional_web_acl_test.go

```diff
@@ -0,0 +1,368 @@
+		_, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+			req := &waf.UpdateWebACLInput{
+				ChangeToken: token,
+				WebACLId:    v.WebACLId,
+			}
+
+			for _, ActivatedRule := range v.Rules {
+				WebACLUpdate := &waf.WebACLUpdate{
+					Action: aws.String("DELETE"),
+					ActivatedRule: &waf.ActivatedRule{
+						Priority: ActivatedRule.Priority,
+						RuleId:   ActivatedRule.RuleId,
+						Action:   ActivatedRule.Action,
+					},
+				}
+				req.Updates = append(req.Updates, WebACLUpdate)
+			}
+
+			return conn.UpdateWebACL(req)
+		})
+		if err != nil {
+			return fmt.Errorf("Error getting change token for waf ACL: %s", err)
+		}
+
+		_, err = wr.RetryWithToken(func(token *string) (interface{}, error) {
+			opts := &waf.DeleteWebACLInput{
+				ChangeToken: token,
+				WebACLId:    v.WebACLId,
+			}
+			return conn.DeleteWebACL(opts)
+		})
+		if err != nil {
+			return fmt.Errorf("Error Deleting WAF ACL: %s", err)
+		}
+		return nil
+	}
+}
+
+func testAccCheckAWSWafRegionalWebAclDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_wafregional_web_acl" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).wafregionalconn
+		resp, err := conn.GetWebACL(
+			&waf.GetWebACLInput{
+				WebACLId: aws.String(rs.Primary.ID),
+			})
+
+		if err == nil {
+			if *resp.WebACL.WebACLId == rs.Primary.ID {
+				return fmt.Errorf("WebACL %s still exists", rs.Primary.ID)
+			}
+		}
+
+		// Return nil if the WebACL is already destroyed
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "WAFNonexistentItemException" {
+				return nil
+			}
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func testAccCheckAWSWafRegionalWebAclExists(n string, v *waf.WebACL) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No WebACL ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).wafregionalconn
+		resp, err := conn.GetWebACL(&waf.GetWebACLInput{
+			WebACLId: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if *resp.WebACL.WebACLId == rs.Primary.ID {
+			*v = *resp.WebACL
+			return nil
+		}
+
+		return fmt.Errorf("WebACL (%s) not found", rs.Primary.ID)
+	}
+}
+
+func testAccAWSWafRegionalWebAclConfig(name string) string {
+	return fmt.Sprintf(`resource "aws_wafregional_ipset" "ipset" {
+  name = "%s"
+  ip_set_descriptors {
+    type = "IPV4"
+    value = "192.0.7.0/24"
+  }
+}
+
+resource "aws_wafregional_rule" "wafrule" {
+  depends_on = ["aws_wafregional_ipset.ipset"]
+  name = "%s"
+  metric_name = "%s"
+  predicates {
+    data_id = "${aws_wafregional_ipset.ipset.id}"
+    negated = false
+    type = "IPMatch"
+  }
+}
+resource "aws_wafregional_web_acl" "waf_acl" {
+  depends_on = ["aws_wafregional_ipset.ipset", "aws_wafregional_rule.wafrule"]
```

Comment:
This is redundant because the relationship is already built by the reference `${aws_wafregional_rule.wafrule.id}` below and further reference inside the rule. Do you mind removing it?